simapp/upgrades.go

```diff
@@ -25,6 +25,17 @@ func (app SimApp) RegisterUpgradeHandlers() {
 	app.UpgradeKeeper.SetUpgradeHandler(
 		UpgradeName,
 		func(ctx context.Context, _ upgradetypes.Plan, fromVM appmodule.VersionMap) (appmodule.VersionMap, error) {
+			// sync accounts and auth module account number
+			currentAccNum, err := app.AuthKeeper.GetAccountNumber(ctx)
```

Comment:
How come we moved this back? Seems like a foot gun for users to remember to put this in

Reply:
since the old auth migration logic run last, when distribution migration run and create a module account which require to fetch the accounts module's acccount number ( by this point is not sync with auth yet since the mirgation haven't run ) so it started with 0. This result in an error `collections: conflict: index uniqueness constrain violation: 0` that i believed is the auth Accounts map index.